core/modules/layout/plugins/access/user_compare_layout_access.inc

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * @file
+ * Contains UserCompareLayoutAccess class.
+ */
+
+/**
+ * Plugin to provide access if two user account contexts are the same.
+ */
+class UserCompareLayoutAccess extends LayoutAccess {
+  /**
+   * Constructor for a Layout access rule.
+   */
+  public function __construct($plugin_name, array $data = array()) {
+    parent::__construct($plugin_name, $data);
+    $this->settings += array(
+      'equality' => 1,
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    $contexts = $this->settings['contexts'];
+    if (empty($contexts) || count($contexts) != 2 || empty($contexts['user1']) || empty($contexts['user2'])) {
+      return t('Compares two user accounts');
+    }
+
+    $labels = array();
+    foreach ($this->settings['contexts'] as $key => $source) {
+      if ($source == 'current_user') {
+        $labels[$key] = t('The logged-in user account');
+      }
+      elseif (is_numeric($source)) {
+        $message = 'The user account from the path (in position @position)';
+        $substitutions = array('@position' => $position = $source + 1);
+        $labels[$key] = t($message, $substitutions);
+      }
+      else {
+        $labels[$key] = t('An unknown user account');
```

Comment:
I find "unknown user account" a bit confusing. Can this be "A user account from the Layout context"

Reply:
I don't think this should ever happen normally? But you're right it's not unknown to the person that set the context, it's just unknown to the compare plugin what kind of user it is. Since it shouldn't occur, any label is probably fine so I don't think we need to waffle on it. I'll change to your suggested string.

Reply:
```suggestion
        $labels[$key] = t('A user account from the layout contexts');
```